### PR TITLE
refactor: new http client; fixed messenger debug method; refactor smapi client

### PR DESCRIPTION
**Pull request by kakhaUrigashvili**
1) Fixed messenger debug method to output debug at the end of a command
2) Created promise based http client with support for ASK_CLI_PROXY env
3) Created smapi client that uses new http client
4) Added refresh access token methods based on new http client

Tests
1) Added/Modified unit tests
2) Some manual tests

debug output
```
ask smapi list-catalogs-for-vendor --debug
ask smapi list-catalogs-for-vendor 
```

http proxy
```
npm i proxy -g
DEBUG=* proxy
# other tab
export ASK_CLI_PROXY=http://localhost:3128
ask smapi list-catalogs-for-vendor 
```

### CODE REVIEW

File: lib/clients/ask-smapi-client/index.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+/**
+ * @param {Object}request Request object
+ * @param {Array<{key, value}>} request.headers Array of headers
+ * @param {string} request.method HTTP method
+ * @param {string} request.url HTTP url
+ * @param {Object} request.body HTTP body
+ */
+const logRequest = (req) => {
+    Messenger.getInstance().info('REQUEST:');
+    Messenger.getInstance().info(jsonView.toString(_stringifyHeaders(req)));
```
Comment (RonWang) [inline lib/clients/ask-smapi-client/index.js:27]: just a head up that this Messenger.info will break the experiences when it's not a pass-through smapi command. Messenger also has debug method, which should be the better one to use here - logs will be flushed at the end of Messenger instance.

Comment (kakhaUrigashvili) [reply lib/clients/ask-smapi-client/index.js:27]: try switching it to Messenger debug - no output

File: lib/clients/ask-smapi-client/index.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    new AppConfig();
+    const authorizationController = new AuthorizationController({
+        auth_client_type: 'LWA',
+        doDebug: debug
+    });
+
+    const refreshTokenConfig = {
+        clientId: authorizationController.oauthClient.config.clientId,
+        clientSecret: authorizationController.oauthClient.config.clientConfirmation,
+        refreshToken: AppConfig.getInstance().getToken(profile).refresh_token
```
Comment (RonWang) [inline lib/clients/ask-smapi-client/index.js:59]: We still need to support the two env var mentioned here https://github.com/alexa/ask-cli/blob/develop/lib/controllers/authorization-controller/index.js#L50-L51

We mentioned the usage in our doc https://github.com/alexa/ask-cli/blob/develop/docs/concepts/CI-CD.md about the existence of those two env var. However, smapi-sdk doesn't support the two env var yet. Should this work be on smapi-sdk?

**Comment by doiron** [comment]
new version of the ask-cli will have a new smapi client

**Pull request closed without merging**